### DISCUSSION TITLE
Time series visualization

### DISCUSSION THREAD

**Discussion by kimjohn1** [SCIInstitute/SCIRun#2259] [Q&A]
SCIRun users

Still a new guy and learning the system.  Can SCIRun render Time series data as an animated view?  Is there an example that I could review, or can you describe the (simplest) set of modules I could use to set up a visualization of a moving animation visualization?

Kim

**Comment by dcwhite** [comment]
GetMatrixSlice can accomplish this, or a python script. Again I reach out to @jessdtate for implementation details...

**Comment by jessdtate** [comment]
can you explain which aspect of the data would be changing in the visualization?  Data that changes on the same mesh is much easier than a changing mesh for instance.  Both would use getmatrixslice, and setfielddata.  

for the case of changing data values, you will need to save a matrix of data with each column representing the data for each time step.  The number of rows will need to match the number of nodes or elements of the mesh.  The transpose is also possible by transposing columns/rows in get matrix slice.  Use getmatrixslice on that matrix to extract on column at a time, then setfielddata to assign it to the mesh.  

for changing meshes, if the node locations are changing and not the connections, that is also relatively easy.  There is a setfieldnode module that will allow you to replace the locations with a matrix of new locations (3xN).  You can also use getmatrixslice, but since there is only one column at a time there are a couple options to use it.  The node matrix can be flattened and combined into a matrix (3*NxM), then reshaped when extracted.  Use ResizeMatrix to flatten and reshape the matrix.  The other option would be to save the X Y Z components as different matrices and combine them (3 NxM matrices).  You would use 3 getmatrixslice modules and connect the index ports so that one of them drive the others.  Then use AppendMatrix to combine the 3 columns into one matrix and setfieldnodes to change the mesh node locations.

If more than the node locations are changing, another strategy would be to save the meshes as separate files with file name that indicates a series, eg, filename_001.fld, filename_002.fld, ....  getmatrixslice and printmatrixintostring can be used iterate through the filenames, then passed into readfield.

**Reply by kimjohn1** [reply]
jessdtate
Keeping things simple, the data is a text file; 26 lines, each line is a set of 3 real values.  The 3 values represent X, Y, Z displacement in a fixed inertial space, with the first position recorded used as the origin point (0,0,0).  The values were generated by a hand held IMU and represent IMU position taken at 1 second intervals.

Here is what I have tried:
Read the data file using AutoReadFile then pass field output to GetFieldData then pass matrix data from GetFieldData to GetMatrixSlice.  Field output from AutoReadFile and matrix output from GetMatrixSlice are passed to SetFieldData, and from there to the usual ShowField and EditMeshBoundingBox, then ViewScene.

The visualization I get is a static view of the complete data set.  I changed the settings in GetMatrixSlice, but those changes had no effect on the visual display.

I can't tell if I'm close to having an animation or way off base.  Let me know if you can tell what I might need to do.  The IMU data file is attached.

Kim
[saved_file.txt](https://github.com/SCIInstitute/SCIRun/files/6322526/saved_file.txt)

**Reply by kimjohn1** [reply]
@jessdtate 
I didn't use the correct addressee syntax.  Please see response above.
Kim
